features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetricDispatcherTest.kt

```diff
@@ -122,29 +133,182 @@ class SessionEndedMetricDispatcherTest {
         val dispatcher = SessionEndedMetricDispatcher(fakeInternalLogger)
 
         // When
-        dispatcher.startMetric(fakeSessionId1, fakeStartReason)
-        dispatcher.startMetric(fakeSessionId2, fakeStartReason)
-        dispatcher.startMetric(fakeSessionId3, fakeStartReason)
-        dispatcher.startMetric(fakeSessionId4, fakeStartReason)
+        dispatcher.startMetric(fakeSessionId1, fakeStartReason, fakeNtpOffsetAtStart, backgroundEventTracking)
+        dispatcher.startMetric(fakeSessionId2, fakeStartReason, fakeNtpOffsetAtStart, backgroundEventTracking)
+        dispatcher.startMetric(fakeSessionId3, fakeStartReason, fakeNtpOffsetAtStart, backgroundEventTracking)
+        dispatcher.startMetric(fakeSessionId4, fakeStartReason, fakeNtpOffsetAtStart, backgroundEventTracking)
 
         // Ends the last started session
-        dispatcher.endMetric(fakeSessionId4)
+        dispatcher.endMetric(fakeSessionId4, fakeNtpOffsetAtEnd)
         // Ends a middle session
-        dispatcher.endMetric(fakeSessionId2)
+        dispatcher.endMetric(fakeSessionId2, fakeNtpOffsetAtEnd)
 
         errorKinds.forEach {
             dispatcher.onSdkErrorTracked(fakeSessionId3, it)
         }
 
         // Then
-        dispatcher.endMetric(fakeSessionId1)
+        dispatcher.endMetric(fakeSessionId1, fakeNtpOffsetAtEnd)
         assertThat(fakeInternalLogger.getErrorCounts()).isEqualTo(0)
 
         // Errors can is reported by the last session when they are tracked
-        dispatcher.endMetric(fakeSessionId3)
+        dispatcher.endMetric(fakeSessionId3, fakeNtpOffsetAtEnd)
         assertThat(fakeInternalLogger.getErrorCounts()).isEqualTo(errorKinds.size)
     }
 
+    @Test
+    fun `M has correct 'with_has_replay' W track multiple views`(
+        @StringForgery fakeSessionId: String,
+        @Forgery viewEvents: List<ViewEvent>
+    ) {
+        // Given
+        val dispatcher = SessionEndedMetricDispatcher(fakeInternalLogger)
+
+        // When
+        dispatcher.startMetric(fakeSessionId, fakeStartReason, fakeNtpOffsetAtStart, backgroundEventTracking)
+
+        viewEvents.map { it.copy(session = it.session.copy(id = fakeSessionId)) }.forEach {
+            dispatcher.onViewTracked(fakeSessionId, it)
+        }
+        dispatcher.endMetric(fakeSessionId, fakeNtpOffsetAtEnd)
+
+        // Then
+        assertThat(fakeInternalLogger.getHasReplayCount()).isEqualTo(viewEvents.count { it.session.hasReplay == true })
+    }
+
+    @Test
+    internal fun `M has correct 'no_view_events_count' W track missed event`(
```

Comment:
unnecessary `internal` visibility modifier

```suggestion
   fun `M has correct 'no_view_events_count' W track missed event`(
```

Reply:
Actually `internal` modifier is mandatory here to not expose internal class `SessionEndedMetric.MissedEventType` since we are using it in the parameter.

Reply:
then it is easier just make the whole test class as `internal`, for the test visibility consistency

Reply:
updated